ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -2187,6 +2188,9 @@ static bool attr_is_modifiable_secret_key(struct pkcs11_attribute_head *attr,
 	case PKCS11_CKA_NEVER_EXTRACTABLE:
 	case PKCS11_CKA_ALWAYS_SENSITIVE:
 		return false;
+	/* CKA_CHECK_VALUE cannot be changed once it has been set. */
+	case PKCS11_CKA_CHECK_VALUE:
```

Comment:
To move to the 1st case above (that returns `true`). The specs says the attribute can be modified: either replaced with a zero-sized value (meaning there is no check value support for the object) or with the valid check-value value.